Sentry.Xamarin.Forms/Internals/NativeIntegration.ios.cs

```diff
@@ -1,11 +1,61 @@
-﻿using Sentry.Integrations;
+﻿using Foundation;
+using Sentry.Integrations;
+using Sentry.Protocol;
+using System;
+using System.Collections.Generic;
+using UIKit;
 
 namespace Sentry.Xamarin.Forms.Internals
 {
     internal partial class NativeIntegration : ISdkIntegration
     {
         internal bool Implemented => true;
-        public void Register(IHub hub, SentryOptions options) { }
-        internal void Unregister() { }
+        private List<NSObject> _observerTokens;
+        private SentryXamarinOptions _xamarinOptions;
+
+        internal NativeIntegration(SentryXamarinOptions options) => _xamarinOptions = options;
+
+        public void Register(IHub hub, SentryOptions options)
+        {
+            _observerTokens = new List<NSObject>();
+            _observerTokens.Add(NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.DidEnterBackgroundNotification, AppEnteredBackground));
+            _observerTokens.Add(NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.WillEnterForegroundNotification, AppEnteredForeground));
+            _observerTokens.Add(NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.DidReceiveMemoryWarningNotification, MemoryWarning));
+        }
+        internal void Unregister()
+        {
+            NSNotificationCenter.DefaultCenter.RemoveObservers(_observerTokens);
+            _xamarinOptions.NativeIntegrationEnabled = false;
+        }
+
+        internal Action<NSNotification> AppEnteredBackground = (_) =>
+        {
+            SentrySdk.AddBreadcrumb(null,
```

Comment:
The goal of passing `hub` to the Register method, is to avoid tight coupling with the static type, which makes it hard for unit testing things

```suggestion
            _hub_.AddBreadcrumb(null,
```